tools/Raven.Debug/LogTrafficWatch/TrafficWatchReplay.cs

```diff
@@ -0,0 +1,313 @@
+            {
+                consumers[i] = new TrafficReplay(_schema, _host, _port).Execute();
+            }
+
+            while (Task.WhenAll(producers).Wait(1_000) == false)
+            {
+                Console.WriteLine($"[{DateTime.UtcNow:O}] Processed {TrafficReplay.RequestsCount:#,#;;0}");
+            }
+
+            TrafficChannel.Writer.Complete();
+
+            await Task.WhenAll(consumers);
+
+            Console.WriteLine($"[{DateTime.UtcNow:O}] Processed {TrafficReplay.RequestsCount:#,#;;0} RequestsCount");
+            Console.WriteLine($"[{DateTime.UtcNow:O}] Done after {sp.Elapsed}");
+        }
+
+        public class TrafficParser
+        {
+            public static int LoadedRequestCount;
+            public static JsonSerializerOptions JsonSerializerOptions = new()
+            {
+                Converters =
+                {
+                    new JsonStringEnumConverter(JsonNamingPolicy.CamelCase),
+                }
+            };
+
+            private readonly string _file;
+
+            public TrafficParser(string file)
+            {
+                _file = file;
+            }
+
+            public async Task Execute()
+            {
+                await foreach (var item in GetItemsList<TrafficWatchChange>(_file))
+                {
+                    await TrafficChannel.Writer.WriteAsync(item);
+                    Interlocked.Increment(ref LoadedRequestCount);
+                }
+            }
+
+            protected virtual async IAsyncEnumerable<T> GetItemsList<T>(string fileName,
+                [EnumeratorCancellation] CancellationToken cancellationToken = default)
+            {
+                await using var fs = new FileStream(fileName, FileMode.Open, FileAccess.Read);
+
+                await foreach (var item in JsonSerializer.DeserializeAsyncEnumerable<T>(fs, JsonSerializerOptions, cancellationToken: cancellationToken))
+                {
+                    if (item != null)
+                        yield return item;
+                }
+            }
+        }
+
+        public class TrafficReplay
+        {
+            private readonly string _schema;
+            private readonly string _host;
+            private readonly int _port;
+
+            public static int RequestsCount;
+
+            public TrafficReplay(string schema = "http", string host = "127.0.0.1", int port = 8080)
+            {
+                _schema = schema;
+                _host = host;
+                _port = port;
+            }
+
+            public async Task Execute()
+            {
+                var responseBuffer = new byte[4 * 1024];
+
+                while (true)
+                {
+                    TrafficWatchChange item;
+                    try
+                    {
+                        item = await TrafficChannel.Reader.ReadAsync();
+                        Interlocked.Increment(ref RequestsCount);
+                    }
+                    catch (Exception)
```

Comment:
let's catch a specific exception here when the channel is completed and empty

Reply:
Done